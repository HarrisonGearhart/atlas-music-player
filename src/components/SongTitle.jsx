import React from "react";

export default function SongTitle({ title, artist }) {
  return (
    <div className="text-center">
      <h2 className="text-lg font-bold text-var(--color-emerald)">{title}</h2>
      <p className="text-sm text-var(--color-azure)">{artist}</p>
    </div>
  );
}
