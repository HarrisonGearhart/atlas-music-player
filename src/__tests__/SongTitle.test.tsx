import React from "react";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import SongTitle from "../components/SongTitle";

describe("SongTitle Component", () => {
  it("renders with default props", () => {
    const { container } = render(<SongTitle />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom title and artist", () => {
    const { container } = render(<SongTitle title="Test Song" artist="Test Artist" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with only title", () => {
    const { container } = render(<SongTitle title="Title Only" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with only artist", () => {
    const { container } = render(<SongTitle artist="Artist Only" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
