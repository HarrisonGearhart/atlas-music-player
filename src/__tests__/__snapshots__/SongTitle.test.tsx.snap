// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > renders with custom title and artist 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Test Song
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Test Artist
  </p>
</div>
`;

exports[`SongTitle Component > renders with default props 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Song Title
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist
  </p>
</div>
`;

exports[`SongTitle Component > renders with only artist 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Song Title
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist Only
  </p>
</div>
`;

exports[`SongTitle Component > renders with only title 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Title Only
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist
  </p>
</div>
`;
